bin/all_sky_search/pycbc_calculate_psd

```diff
@@ -48,15 +48,29 @@ def get_psd(input_tuple):
     argstmp.gps_start_time = int(seg[0]) + args.pad_data
     argstmp.gps_end_time = int(seg[1]) - args.pad_data
     tmp_segment = segment([argstmp.gps_start_time, argstmp.gps_end_time])
-    argstmp.channel_name = resolve_td_option(args.channel_name, tmp_segment)
+    # There may be cases where channel name or frame type change within an
+    # analysis (noting that this could be an issue with significance
+    # computation). If so, resolve them for each PSD block. The channel or
+    # frame type *cannot* change during a lock stretch.
+    if args.channel_name:
+        # Option may not have been given at all
+        argstmp.channel_name = resolve_td_option(args.channel_name, tmp_segment)
+    if args.frame_type:
+        # Option is often not given.
+        argstmp.frame_type = resolve_td_option(args.frame_type, tmp_segment)
 
     # This helps when the filesystem is unreliable, and gives extra retries.
     # python has an internal limit of ~100 (it is not infinite)
     try:
         gwstrain = pycbc.strain.from_cli(argstmp, pycbc.DYN_RANGE_FAC)
     except RuntimeError:
+        # Retry on failures ... but don't keep trying forever!!
+        if rerun_count is None:
+            rerun_count = 0
+        if rerun_count == 5:
+            raise
         time.sleep(10)
-        return get_psd((seg, i))
+        return get_psd((seg, i), rerun_count=rerun_count+1)
```

Comment:
I think a for loop would be simpler and clearer than recursion here, and it would allow you to redefine the `rerun_count` argument to actually mean how many times you are willing to try, rather than having an hardcoded max.

Reply:
Also, the function would behave oddly if called with `rerun_count = 6` or larger.

Reply:
You can also remove the `rerun_count` arg now.